bifrost/app/blog/blogs/debugging-chatbots-and-ai-agents-with-sessions/src.mdx

```diff
@@ -0,0 +1,171 @@
+
+
+**⚠️ Lack of Visibility into Internal States**
+
+Without proper tools, the internal workings of an AI agent can be a "black box," with **<span style={{color: '#0ea5e9'}}>little visibility into how inputs are transformed into outputs</span>**. In systems where agents interact with multiple services or other agents, tracing the flow of information becomes even more challenging. In addition, traditional logging methods may not capture the necessary granularity of data needed to debug complex AI behaviors effectively.
+
+**⚠️ Contextual Debugging Challenges**
+
+AI agents often operate over extended interactions or sessions, where the context accumulates over time. Tracking the agent's state throughout a session is essential for debugging since **<span style={{color: '#0ea5e9'}}>an error early in a session can have cascading effects</span>**, making it difficult to pinpoint the original source without comprehensive session data.
+
+
+While understanding the internal workings of complex AI models is inherently challenging, using tools like Helicone’s Session significantly alleviate these difficulties. 
+
+
+---
+
+## What are Sessions?
+
+**Sessions** provide a simple way to organize and group related LLM calls, making it easier for developers to trace nested agent workflows and visualize interactions between the user and the AI chatbot or agent. 
+
+Instead of looking at isolated data points, sessions allow you to: 
+- see a comprehensive view of an entire conversation or interactive flow.
+- drill down on specific LLM calls to view your agent’s flow of task execution.
+- identify issues quicker given a better understanding of the context of errors.
+- refine your chatbot's responses based on specific contexts.
+
+
+## Using Sessions in Helicone
+
+1. Simply add `Helicone-Session-Id` to start tracking your sessions. 
+2. Add `Helicone-Session-Path`to specific parent and child traces. 
+
+Here is an example in **TypeScript**: 
+
+```ts
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY,
+  baseURL: "https://oai.helicone.ai/v1",
+  defaultHeaders: {
+    "Helicone-Auth": `Bearer ${process.env.HELICONE_API_KEY}`,
+  },
+});
+
+const session = randomUUID();
+
+openai.chat.completions.create(
+  {
+    messages: [
+      {
+        role: "user",
+        content: "Generate an abstract for a course on space.",
+      },
+    ],
+    model: "gpt-4",
+  },
+  {
+    headers: {
+      "Helicone-Session-Id": session, // defines your session
+      "Helicone-Session-Path": "/abstract", // defines the order of LLM calls
+    },
+  }
+);
+```
+
+
+## How Industries Use Sessions to Debug AI Agents 
+
+### Resolving Errors in Multi-Step Processes 
```

Comment:
Adding images of these sessions within Helicone would be awesome.

potentially they could be made in figma vs coding them